### ISSUE TITLE
Question: Why are the Docksal boilerplates different from each other and the drupal.org recommended project regarding docroot name and config folder setup?

### ISSUE THREAD

**Issue by zhouhana** [docksal/boilerplate-drupal9-blt#6]
I was comparing the Drupal recommended project template with the old drupal-composer template and the three Docksal boilerplates, and I notices that drupal9 and drupal9-blt (but not drupal9-composer) names the DOCROOT folder 'docroot' instead of the Drupal default 'web'. Is there a reason for this? If not, it feels to me like it would be good practice to keep the setup as similar as possible to the Drupal default/recommended project, and thus rename it 'web' in drupal9 and drupal-blt. Anything else will only cause unnecessary questions from Composer noobs like myself. (Or perhaps you feel strongly that it should always be 'docroot'? If so, it would be good to propose this change to the recommended project maintainers and also to change 'web' to 'docroot' in drupal9-composer.)

Another thing I noticed is that the drupal9-blt boilerplate, along with the drupal9 boilerplate (but not the drupal9-composer boilerplate), does place a 'config' directory outside of the webroot, which is great, but unlike the drupal9 boilerplate, drupal9-blt doesn't add a config sync path to it in settings.php. Compare with the settings file from the drupal9 boilerplate:

<img width="319" alt="Screenshot 2021-07-02 at 15 56 53" src="https://user-images.githubusercontent.com/493629/124285440-351da780-db4e-11eb-9709-367ed680a088.png">

In other words, all three Docksal boilerplates do this more or less differently. If possible I think that both drupal9, drupal9-composer and drupal9-blt should have the same setup regarding the config directory.

**Comment by shelane** [comment]
The [BLT project](https://docs.acquia.com/blt/) and the [Acquia Drupal Minimal Project](https://github.com/acquia/drupal-minimal-project) use docroot as the web root. This is because that is what the Acquia infrastructure is set up for. Other service providers use "web" as the default. The newly created [Drupal Recommended Project](https://github.com/drupal/recommended-project) (as of Drupal 8.6 I believe) uses "web" as the default. Our boilerplate projects follow each of these based on their defaults.

**Comment by lpeabody** [comment]
> and I notices that drupal9 and drupal9-blt (but not drupal9-composer) names the DOCROOT folder 'docroot' instead of the Drupal default 'web'. Is there a reason for this?

As @shelane mentioned, BLT is an Acquia product, and Acquia is standardized on having `docroot` as the Drupal root, rather than `web`.

> does place a 'config' directory outside of the webroot, which is great, but unlike the drupal9 boilerplate, drupal9-blt doesn't add a config sync path to it in settings.php.

In the BLT boilerplate, in sites/default/settings.php there is a file included:

```
require DRUPAL_ROOT . "/../vendor/acquia/blt/settings/blt.settings.php";
```

Where, if you look through, `blt.settings.php` includes another file `config.settings.php` which sets `$settings['config_sync_directory']`.

BLT assumes a lot because of its convention-first approach. Check out the documentation for more info. https://docs.acquia.com/blt/

Hope that clears up any confusion, @zhouhana.

**Comment by zhouhana** [comment]
Thank you both! That explains everything, except why web is renamed docroot in the drupal9 blueprint. I guess that question is better answered in that repo, though.
